sql/src/main/java/io/crate/analyze/relations/RelationAnalysisContext.java

```diff
@@ -51,11 +52,22 @@ public ParameterContext parameterContext() {
     }
 
     public void addSourceRelation(String nameOrAlias, AnalyzedRelation relation) {
-        sources.put(new QualifiedName(nameOrAlias), relation);
+        if (sources.put(new QualifiedName(nameOrAlias), relation) != null) {
```

Comment:
can't we use addSourceRelation(String schemaName, String nameOrAlias, AnalyzedRelation relation) here to avoid duplicating error logic?